File: Java/Arrays/leftRotateByDPlaces.java
import java.util.*;

class leftRotateByDPlaces{

    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the size of the array : ");
        int n = sc.nextInt();
        ArrayList<Integer> arr = new ArrayList<>();
        System.out.println("Enter the elements of the array : ");
        for(int i=0;i<n;i++){
            int k = sc.nextInt();
            arr.add(k);
        }
        System.out.print("Enter the no of places to rotate : ");
        int d = sc.nextInt();
        d = d % n;
        leftRotateByD(arr, d);
        System.out.println("Array after rotating by D places : " + arr);
    }

    public static void leftRotateByD(ArrayList<Integer> arr, int d){
        rotate(arr, 0, d-1);
        rotate(arr, d, arr.size()-1);
        rotate(arr, 0, arr.size()-1);
    }

    public static void rotate(ArrayList<Integer> arr, int l ,int h){
        int temp = 0;
        while(l < h){
            temp = arr.get(l);
            arr.set(l, arr.get(h));
            arr.set(h, temp);
            l++;
            h--;
        }
    }
};

File: Python/Arrays/leftRotateByDPlaces.py
def rotate(arr, l, h):
    temp = 0
    while(l < h):
        temp = arr[l]
        arr[l] = arr[h]
        arr[h] = temp
        l = l + 1
        h = h - 1

def leftRotateByD(arr, d):
    rotate(arr, 0, d-1)
    rotate(arr, d, len(arr)-1)
    rotate(arr, 0, len(arr)-1)

if __name__ == "__main__":
    n = int(input("Enter the size of the array : "))
    arr = list()
    print("Enter the elements of the array : ")
    for i in range(0, n):
        k = int(input())
        arr.append(k)
    print("Enter the no of places to rotate : ")
    d = int(input())
    d = d % n
    leftRotateByD(arr, d)
    print("Array after rotation : ", arr)